unified-runtime/cmake/FetchLevelZero.cmake

```diff
@@ -3,146 +3,139 @@
+
+  # Restore original flags
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS_BAK}")
+  set(CMAKE_MSVC_RUNTIME_LIBRARY "${CMAKE_MSVC_RUNTIME_LIBRARY_BAK}")
+  set(LEVEL_ZERO_LIBRARY ze_loader)
+  set(LEVEL_ZERO_INCLUDE_DIR
+    ${level-zero-loader_SOURCE_DIR}/include CACHE PATH "Path to Level Zero Headers")
+
+  set(LEVEL_ZERO_TARGET_INCLUDE_DIR "${CMAKE_INSTALL_INCLUDEDIR}")
+
+  file(GLOB LEVEL_ZERO_LOADER_API_HEADERS "${LEVEL_ZERO_INCLUDE_DIR}/*.h")
+  file(COPY ${LEVEL_ZERO_LOADER_API_HEADERS} DESTINATION ${LEVEL_ZERO_INCLUDE_DIR}/level_zero)
+
+  target_compile_options(ze_loader PRIVATE
+    $<$<IN_LIST:$<CXX_COMPILER_ID>,GNU;Clang;Intel;IntelLLVM>:-Wno-error>
+    $<$<CXX_COMPILER_ID:MSVC>:/WX- /UUNICODE>
     )
-
-    set(LEVEL_ZERO_LIBRARY ze_loader)
-    set(LEVEL_ZERO_INCLUDE_DIR
-        ${level-zero-loader_SOURCE_DIR}/include CACHE PATH "Path to Level Zero Headers")
 endif()
 
 add_library(LevelZeroLoader INTERFACE)
 # The MSVC linker does not like / at the start of a path, so to work around this
 # we split it into a link library and a library path, where the path is allowed
 # to have leading /.
+if(NOT LEVEL_ZERO_LIBRARY_SRC OR NOT LEVEL_ZERO_LIB_NAME)
 get_filename_component(LEVEL_ZERO_LIBRARY_SRC "${LEVEL_ZERO_LIBRARY}" DIRECTORY)
 get_filename_component(LEVEL_ZERO_LIB_NAME "${LEVEL_ZERO_LIBRARY}" NAME)
+endif()
+
+if(NOT LEVEL_ZERO_TARGET_INCLUDE_DIR)
+  set(LEVEL_ZERO_TARGET_INCLUDE_DIR ${LEVEL_ZERO_INCLUDE_DIR})
+  endif()
+
 target_link_directories(LevelZeroLoader
     INTERFACE "$<BUILD_INTERFACE:${LEVEL_ZERO_LIBRARY_SRC}>"
-              "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
+              "$<INSTALL_INTERFACE:${LEVEL_ZERO_TARGET_INCLUDE_DIR}>"
 )
 target_link_libraries(LevelZeroLoader
     INTERFACE "${LEVEL_ZERO_LIB_NAME}"
 )
 
-file(GLOB LEVEL_ZERO_LOADER_API_HEADERS "${LEVEL_ZERO_INCLUDE_DIR}/*.h")
-file(COPY ${LEVEL_ZERO_LOADER_API_HEADERS} DESTINATION ${LEVEL_ZERO_INCLUDE_DIR}/level_zero)
 add_library(LevelZeroLoader-Headers INTERFACE)
 target_include_directories(LevelZeroLoader-Headers
-    INTERFACE "$<BUILD_INTERFACE:${LEVEL_ZERO_INCLUDE_DIR};${LEVEL_ZERO_INCLUDE_DIR}/level_zero>"
-              "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
+    INTERFACE "$<BUILD_INTERFACE:${LEVEL_ZERO_INCLUDE_DIR}>"
+              "$<INSTALL_INTERFACE:${LEVEL_ZERO_TARGET_INCLUDE_DIR}>"
 )
-
 find_path(L0_COMPUTE_RUNTIME_HEADERS
   NAMES "ze_intel_gpu.h"
   PATH_SUFFIXES "level_zero"
 )
-if(NOT UR_COMPUTE_RUNTIME_REPO AND L0_COMPUTE_RUNTIME_HEADERS)
+if(L0_COMPUTE_RUNTIME_HEADERS)
     set(COMPUTE_RUNTIME_LEVEL_ZERO_INCLUDE "${L0_COMPUTE_RUNTIME_HEADERS}")
     set(COMPUTE_RUNTIME_REPO_PATH "${L0_COMPUTE_RUNTIME_HEADERS}")
-elseif (UR_COMPUTE_RUNTIME_FETCH_REPO)
-    # Fetch only if UR_COMPUTE_RUNTIME_FETCH_REPO is set to ON.
-    if (UR_COMPUTE_RUNTIME_REPO STREQUAL "")
-        set(UR_COMPUTE_RUNTIME_REPO "https://github.com/intel/compute-runtime.git")
-    endif()
-    if (UR_COMPUTE_RUNTIME_TAG STREQUAL "")
```

Comment:
If we remove these, we won't be able to `overwrite` the REPO/TAG using cmake command line defines, which was useful when doing debugging without having to *hack* the cmake file just for debugging.